eli5/_feature_names.py

```diff
@@ -110,6 +117,20 @@ def filtered_by_re(self, feature_re):
             ),
             indices)
 
+    def handle_filter(self, feature_filter, feature_re, x=None):
+        if feature_re is not None and feature_filter:
+            raise TypeError('pass either feature_filter or feature_re')
```

Comment:
I think ValueError is more common for such cases (see e.g. http://stackoverflow.com/questions/256222/which-exception-should-i-raise-on-bad-illegal-argument-combinations-in-python?rq=1)